test/TestProjects/SerializationCustomization/Generated/Models/EmptyAsUndefinedTestModel.Serialization.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Linq;
+using System.Text.Json;
+using Azure.Core;
+
+namespace SerializationCustomization.Models
+{
+    public partial class EmptyAsUndefinedTestModel : IUtf8JsonSerializable
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            if (EmptyAsUndefinedList != null && EmptyAsUndefinedList.Any())
```

Comment:
`Any()` in .NET Framework is slow and allocates a new enumerator. In .NET Core, it checks for `ICollection<T>` or `ICollection` first to use its `Count` property. As long as you know this will be `IList<T>` or `IReadOnlyList<T>`, just use it's `Count` property for faster/less allocations code.

Reply:
Don't want to add a complicated type mapping logic yet. I'll do it if it becomes a problem.

Reply:
How about this, then (I did this for the search engine): write our own extension method like `IsEmpty()` (I typically do `IsEmptyOrNull`, but whichever) that does what .NET Core does, which is basically:

```csharp
if (obj is ICollection<T> c) return c.Count > 0;
else if (object is ICollection c) return c.Count > 0;
else { var e = obj.GetEnumerator(); return !e.MoveNext(); }
```

Still, maybe not worth it (especially when you consider Framework code is NGEN'd, which makes a different if called a lot (I tested)). Just wanted to call it out as an option that works in both cases.

Reply:
It would definitely work and would definitely have an effect if called a lot, just don't know if we are there yet.